javascript-algorithms-and-data-structures/learn-basic-string-and-array-methods-by-building-a-music-playlist-app/scripts.js

```diff
@@ -0,0 +1,202 @@
+    playSong(songs[currentIndex]);
+  } else {
+    pauseSong();
+  }
+
+  updatePlayingSongBkg(currentIndex)
+
+}
+
+function toggleClosePlaylist() {
+    if (songListDiv.style.visibility == "hidden") {
+        songListDiv.style.visibility = "visible";
+        songListDiv.style.display = "flex";
+
+    } else {
+        songListDiv.style.visibility = "hidden";
+        songListDiv.style.display = "none";
+    }
+}
+
+function currentSong() {
+  return songs.find((song) => {
+    song.src === audio.src
+}) || songs[currentIndex];
+}
+
+// player controls
+function nextSong() {
+  const nextIndex = (currentIndex + 1) % songs.length;
+  playSong(songs[nextIndex]);
+  currentIndex = nextIndex;
+}
+
+function previousSong() {
+  const prevIndex = (currentIndex - 1 + songs.length) % songs.length;
+  playSong(songs[prevIndex]);
+  currentIndex = prevIndex;
+}
+
+// playlist display
+function clearSongBgs() {
+    document.querySelectorAll(".playlist__song").forEach(song => {
+        song.classList.remove('selected');
+    });
+}
+
+function updatePlayingSongBkg(newIndex) {
+    const songDivs = document.querySelectorAll(".playlist__song");
+    songDivs[newIndex].classList.add('selected');
+}
+
+
+function playSelectedSong(song, songDiv) {
+    playSong(song);
+    clearSongBgs();
+
+    songDiv.classList.add('selected');
+}
+
+
+function shuffle() {
+    // Shallow copy done with spread operator
+    let songsCopy = [...songs];
+    // using array sort method and Math.random method
+    songsCopy.sort(() => 0.5 - Math.random());
+    renderSongs(songsCopy);
+    playSong(songsCopy[currentIndex])
+}
+
+// Event listeners
+document.querySelector(".play").addEventListener("click", togglePlayPause);
+document.querySelector(".previous").addEventListener("click", previousSong);
+document.querySelector(".next").addEventListener("click", nextSong);
+document.querySelector(".shuffle").addEventListener("click", shuffle);
+
+
+document.querySelector(".playlist__close").addEventListener("click", toggleClosePlaylist);
+
+
+// continue playing next song
+audio.addEventListener("ended", nextSong);
+
+
+// keyboardEvents
+window.addEventListener("keydown", (event) => {
+//     if (event.key === "ArrowUp") {
+//         previousSong();
+//     } else if (event.key === "ArrowDown") {
+//         nextSong();
+//     } else if (event.key == " ") {
+//        togglePlayPause();
+//    }
+
+    switch (true) {
+        case (event.key === "ArrowUp"):
+            previousSong();
+            break
+        case (event.key === "ArrowDown"):
+            nextSong();
+            break
+        case (event.key === " "):
+            event.preventDefault();
+            togglePlayPause();
+            break
+    }
+})
+
+
+document.querySelectorAll('div[role="button"]').forEach((songDiv,index) => {
+    songDiv.addEventListener('keydown', event => {
+        if (event.key === 'Enter') {
+            playSelectedSong(songs[index], songDiv);
+        }
+    });
+});
```

Comment:
```suggestion
document.querySelectorAll('button.playlist__song').forEach((song, index) => {
    song.addEventListener('click', event => playSelectedSong(songs[index], song));
});
```

Now that we are using actual `button`s for the songs in the list we can just use a click handler and now the Enter key, Spacebar, and any pointer clicks will start the song playing.